cprover_bindings/src/goto_program/typ.rs

```diff
@@ -104,6 +146,12 @@ pub struct Parameter {
     base_name: Option<InternedString>,
 }
 
+/// Type for function contracts, loop contracts, etc.
+#[derive(Debug, Clone)]
+pub enum Contract {
```

Comment:
Does this need to be an enum?

Reply:
Made it an enum to support different types of contracts. Like `LoopContract(...)` in the future. Is there an alternative way to do this?